libs/pipedream.libsonnet

```diff
@@ -153,14 +177,23 @@ local pipedream_rollback_pipeline(pipedream_config, service_pipelines, trigger_p
 
 // generate_region_pipeline will call the pipeline callback function, and then
 // name the pipeline, add an upstream material, and append a final stage.
-local generate_region_pipeline(pipedream_config, region, weight, pipeline_fn) =
-  // Get previous region's pipeline name
+// pipedream_config: The configuration passed into the render() function
+// pipeline_fn:      The callback function passed in to render() function
+// region:           The region to create pipelines for
+// display_order:    The order of the pipeline in GoCD UI
+local generate_region_pipeline(pipedream_config, pipeline_fn, region, display_order) =
   local service_name = pipedream_config.name;
+  local service_pipeline = pipeline_fn(region);
 
-  local service_pipeline = pipeline_fn(
-    region,
-  );
-
+  // `auto_pipeline_progression` was added as a utility for folks new to
+  // pipedream. When this is false, each region will need manual approval
+  // before doing any of the deployment stages.
+  // We add ready + wait stages to improve the GoCD UI since it will start
+  // the regions pipeline, show a green check and then a manual approval arrow.
+  // If the first stage had manual approval, GoCD assumes the pipeline itself
+  // needs manual approval and expects the user to manually trigger the
+  // pipeline through the play+ icon, which isn't clear where in pipedream a
+  // deploy is.
```

Comment:
I am still a bit confused about what this means

Reply:
I've tried to simplify the comment, let me know what you think

Reply:
Honestly, after re-reading this, it makes more sense. I feel that it may be better to be more verbose here, so I am now leaning toward the original. Sorry about that 😅